tests/test_queries.py

```diff
@@ -0,0 +1,890 @@
+        non_value_columns.update(self._model.supplemental_columns)
+        value_columns = sorted((x for x in df.columns if x not in non_value_columns))
+        expected = ["electricity_cooling", "electricity_heating"]
+        success = value_columns == expected
+        if not success:
+            logger.error("Mismatch in columns: actual=%s expected=%s", value_columns, expected)
+        if set(self._model.supplemental_columns).difference(df.columns):
+            logger.error(
+                "supplemental_columns=%s are not present in table",
+                self._model.supplemental_columns,
+            )
+            success = False
+        if not df.select("county").distinct().filter(f"county == '{county_name}'").collect():
+            logger.error("County name = %s is not present", county_name)
+            success = False
+        if success:
+            total_cooling = df.agg(F.sum("electricity_cooling").alias("sum")).collect()[0].sum
+            total_heating = df.agg(F.sum("electricity_heating").alias("sum")).collect()[0].sum
+            expected = self.get_raw_stats()["by_county"][county]["comstock_resstock"]["sum"]
+            if not math.isclose(total_cooling, expected["electricity_cooling"]):
+                logger.error(
+                    "Mismatch in electricity_cooling: actual=%s expected=%s",
+                    total_cooling,
+                    expected["electricity_cooling"],
+                )
+                success = False
+            if not math.isclose(total_heating, expected["electricity_heating"]):
+                logger.error(
+                    "Mismatch in electricity_heating: actual=%s expected=%s",
+                    total_heating,
+                    expected["electricity_heating"],
+                )
+                success = False
+        return success
+
+
+class QueryTestElectricityUseByCounty(QueryTestBase):
+
+    NAME = "electricity-use-by-county"
+
+    def make_query(self):
+        self._model = ProjectQueryResultModel(
+            name=self.NAME,
+            project=ProjectConstraintsModel(
+                project_id="dsgrid_conus_2022",
+                include_dsgrid_dataset_components=True,
+                dataset_ids=[
+                    "conus_2022_reference_comstock",
+                    "conus_2022_reference_resstock",
+                    # "tempo_conus_2022",
+                ],
+                metric_reductions=[
+                    MetricReductionModel(dimension_query_name="electricity", operation="sum")
+                ],
+            ),
+            aggregations=[
+                AggregationModel(
+                    group_by_columns=["county"],
+                    aggregation_function=F.sum,
+                    name="sum",
+                ),
+                AggregationModel(
+                    group_by_columns=["county"],
+                    aggregation_function=F.max,
+                    name="max",
+                ),
+            ],
```

Comment:
Currently, it looks like this just create two output files under the "electricity-use-by-county" query dir -- one for max and one for sum. These do not look chained (which you mentioned as something not being implemented yet). So does this mean that these are independent queries under the same root query name?

Reply:
Yes

Reply:
Is this something we want to encourage? Or is this just a placeholder for when we get to chained aggregations?

Reply:
In a past meeting we agreed that we should support independent aggregations, and so I implemented it. Isn't this something a user would want to do? Like find the max load by county and total load by county?